idaes/core/util/parameter_sweep.py

```diff
@@ -0,0 +1,655 @@
+
+        space_init = self.sampling_method(
+            bounds_list,
+            sample_size,
+            sampling_type="creation",
+        )
+
+        self._samples = DataFrame(
+            space_init.sample_points(), columns=[_ for _ in self.inputs.keys()]
+        )
+
+        return self.samples
+
+    @property
+    def inputs(self):
+        return self._inputs
+
+    @property
+    def sampling_method(self):
+        return self._sampling_method
+
+    @property
+    def sample_size(self):
+        return self._sample_size
+
+    @property
+    def samples(self):
+        return self._samples
+
+    def to_dict(self):
+        """Write specification to a dict
+
+        Returns:
+            Input specification as a dict
+        """
+        outdict = {}
+
+        outdict["inputs"] = self._inputs
+
+        # Pysmo sampling methods are not serializable, so get name instead
+        outdict["sampling_method"] = self._sampling_method.__name__
+        outdict["sample_size"] = self._sample_size
+        outdict["samples"] = self._samples.to_dict(orient="tight")
+
+        return outdict
+
+    def from_dict(self, input_dict: dict):
+        """
+        Load specification from dict
+
+        Args:
+            input_dict: dict to load into specification
+
+        Returns:
+            None
+        """
+        self._inputs = OrderedDict()
+        for k, v in input_dict["inputs"].items():
+            self._inputs[k] = OrderedDict(v)
+
+        # Hack to get Pysom sampling method from string name
+        mod = __import__(
+            "idaes.core.surrogate.pysmo.sampling",
+            fromlist=[input_dict["sampling_method"]],
+        )
+        self._sampling_method = getattr(mod, input_dict["sampling_method"])
+
+        self._sample_size = input_dict["sample_size"]
+        self._samples = DataFrame().from_dict(
+            input_dict["samples"],
+            orient="tight",
+        )
+
+    def to_json_file(self, filename: str):
+        """
+        Serialize specification to file in json format.
+
+        Args:
+            filename: name of file to write to as string
+
+        Returns:
+            None
+        """
+        with open(filename, "w") as fd:
+            json.dump(self.to_dict(), fd, indent=3)
+
+    def from_json_file(self, filename: str):
+        """
+        Load specification from json file.
+
+        Args:
+            filename: name of file to load as string
+
+        Returns:
+            None
+        """
+        with open(filename, "r") as f:
+            self.from_dict(json.load(f))
+        f.close()
+
+
+def is_psweepspec(val):
+    if isinstance(val, ParameterSweepSpecification):
+        return val
+    _log.error(
+        f"Input configuration {val} must be an instance of ParameterSweepSpecification."
+    )
+    raise ValueError(
+        "Input configuration must be an instance of ParameterSweepSpecification."
+    )
+
+
+CONFIG = ConfigDict()
+CONFIG.declare(
+    "build_model",
```

Comment:
in watertap, the build step is parameterized (ie takes keywords, the name/desc of which is declared per-flowsheet) to reuse models more easily (eg stages in RO or treatment options for output streams)